tools/build/build.sh

```diff
@@ -50,7 +50,10 @@ if (( $# > 0 )); then
   readonly GIT_ROOT=$(git rev-parse --show-toplevel)
   # Rather than a working directory of something like "/worker", mirror
   # the path on the host so that symlink tricks work as expected.
-  docker run --rm -ti -v "$GIT_ROOT":"$GIT_ROOT" -w "$GIT_ROOT" $IMAGE_NAME "$@"
+  # We cache ~/.gradle between builds, for speed.  We also ensure the target cache folder exists.
+  readonly GRADLE_CACHE_DIR="$GIT_ROOT/build/.gradle-docker"
+  mkdir -p "$GRADLE_CACHE_DIR"
+  docker run --rm -ti -v "$GIT_ROOT":"$GIT_ROOT" -w "$GIT_ROOT" -v "$GRADLE_CACHE_DIR:/root/.gradle" $IMAGE_NAME "$@"
```

Comment:
Just some notes: When moving from `/tmp` to `build` I had to address two concerns:

- If the `build` folder does not yet exist, I fear docker will fail to mount `build/.gradle-docker`, so I added a `mkdir -p`

  (Strictly speaking, creating just the parent folder `build` would have been enough, but this way better communicates the intention.)

- There is a danger than `$GIT_ROOT` may contain spaces, so I had to bring the `-v` option directly inline.